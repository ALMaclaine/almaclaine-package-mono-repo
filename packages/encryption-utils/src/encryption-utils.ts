import {hash, compare} from 'bcrypt';

export async function encryptPassword(password: string, rounds: number = 64000) {
    return hash(password, rounds);
}

export async function comparePassword(providedPassword: string, actualPassword: string) {
    return compare(providedPassword, actualPassword);
}
