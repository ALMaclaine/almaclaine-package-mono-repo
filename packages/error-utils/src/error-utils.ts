export class ALMError extends Error {
    constructor(message: string, public readonly type: string) {
        super(message);
    }
}
