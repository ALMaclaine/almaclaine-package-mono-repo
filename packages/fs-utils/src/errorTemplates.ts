export const doesNotExistError = (path: string): string => `Provided path ${path} does not exist: ${path}.`;

export const alreadyExistsError = (path: string): string => `Provided file/directory already exists: ${path}.`;

export const isNotDirectoryError = (path: string): string => `Provided path ${path} is not a directory.`;

export const isDirectoryError = (path: string): string => `Provided path ${path} is a directory.`;

export const isNotJsonFilError = (path: string): string => `Provided path ${path} is not a json file. Extension must be .json`;

