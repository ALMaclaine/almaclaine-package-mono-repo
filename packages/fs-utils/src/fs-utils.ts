import {lstatSync, existsSync, readdirSync, readFileSync as fsReadFileSync, writeFileSync, mkdirSync, createReadStream as fsCreateReadStream} from 'fs';
import {join, sep} from 'path';
import {
    alreadyExistsError,
    doesNotExistError,
    isDirectoryError,
    isNotDirectoryError,
    isNotJsonFilError
} from "./errorTemplates";
import {Stream} from "stream";

export {join, resolve} from 'path';
export {existsSync, copyFileSync} from 'fs';

export function isHidden(path: string): boolean {
    const pathSplit = path.split(sep);
    return pathSplit[pathSplit.length - 1][0] === '.';
}

export function createReadStream(path: string): Stream {
    if(!existsSync(path)) {
        throw Error(doesNotExistError(path));
    }

    return fsCreateReadStream(path);
}

export const makeCwdRelPath = (...files: Array<string>): string => join(process.cwd(), ...files);

export function isDirectorySync(path: string): boolean {
    if(!existsSync(path)) return false;
    return lstatSync(path).isDirectory();
}

export function readDirectorySync(path: string): Array<string> {
    if(!isDirectorySync(path)) {
        throw Error(isNotDirectoryError(path));
    }

    return readdirSync(path);
}

export function makeDirectorySync(path: string): void {
    if(existsSync(path)) {
        throw Error(alreadyExistsError(path));
    }
    mkdirSync(path);
}

export function importJson(path: string): any {
    if(!existsSync(path)) {
        throw Error(doesNotExistError(path));
    }

    if(!/.*\.json/.test(path)) {
        throw Error(isNotJsonFilError(path));
    }

    return require(path);
}

export function readFileAsStringSync(path: string): string {
    if(!existsSync(path)) {
        throw Error(doesNotExistError(path));
    }

    if(isDirectorySync(path)) {
        throw Error(isNotDirectoryError(path));
    }

    return fsReadFileSync(path, 'utf-8');
}

export function readFileSync(path: string): Buffer {
    if(!existsSync(path)) {
        throw Error(doesNotExistError(path));
    }

    if(isDirectorySync(path)) {
        throw Error(isNotDirectoryError(path));
    }

    return fsReadFileSync(path);
}

export function writeFileAsStringSync(path: string, data: string): void {
    if(isDirectorySync(path)) {
        throw Error(isDirectoryError(path));
    }

    writeFileSync(path, data, 'utf-8');
}
