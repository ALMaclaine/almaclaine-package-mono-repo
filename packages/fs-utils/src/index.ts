export * from './fs-utils';
export * from './errorTemplates';
