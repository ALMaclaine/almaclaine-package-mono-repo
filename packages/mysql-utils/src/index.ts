export * from './mysql-utils';
export type {ConnectionInfo} from './types';
