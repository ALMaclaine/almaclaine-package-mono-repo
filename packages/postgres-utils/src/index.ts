export * from './postgres-utils';
export * from './ConnectionManager';
