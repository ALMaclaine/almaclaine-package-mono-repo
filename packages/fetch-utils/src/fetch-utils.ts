const fetch = require('cross-fetch');
import {ALMError} from "@almaclaine/error-utils";

export const ErrorTypes = {
    FetchFailed: 'FetchFailed'
}

export class FetchFailedError extends ALMError {
    constructor(message: string) {
        super(message, ErrorTypes.FetchFailed);
    }
}

export async function almFetch(url: string, options: {[key: string]: any} = {}) {
    const res = await fetch(url, options);
    if(!res.ok) {
        throw new FetchFailedError(`Fetch failed with status code: ${res.status}`);
    }
    return res;
}
