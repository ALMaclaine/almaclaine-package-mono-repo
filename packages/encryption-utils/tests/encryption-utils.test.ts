import * as encryptionUtils from '../src';

describe('encryption-utils', () => {
    it('Has Correct API', () => {
        const keys = Object.keys(encryptionUtils);
        expect(keys.length).toBe(2);
        expect(keys).toMatchSnapshot();
    });
});
