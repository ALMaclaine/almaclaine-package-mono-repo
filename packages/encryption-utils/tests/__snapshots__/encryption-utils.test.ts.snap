// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`encryption-utils Has Correct API 1`] = `
Array [
  "encryptPassword",
  "comparePassword",
]
`;
